### PR TITLE
fix: Allow for searching TechDocs by Title

### PR DESCRIPTION
**Pull request by CptnFizzbin**
Came across an issue where trying to find a TechDoc in the index table by title wasn't showing up. 

The Entity table was updated to allow for searching by title in #12275, but the same change didn't make it over to TechDocs.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by Wilson-StephenA** [comment]
Note: Change to @backstage/plugin-techdocs-backend only includes adding a new example component for testing

File: plugins/techdocs/src/home/components/Tables/columns.tsx [hunk 1]
```diff
@@ -30,11 +30,29 @@ function customTitle(entity: Entity): string {
  * @public
  */
 export const columnFactories = {
+  createTitleColumn(options?: { hidden?: boolean }): TableColumn<DocsTableRow> {
+    return {
+      title: 'Document',
```
Comment (squid-ney) [inline plugins/techdocs/src/home/components/Tables/columns.tsx:35]: Is it intentional that `title` for `createTitleColumn` is the same as the `createNameColumn`? Both are 'Document'

Comment (Wilson-StephenA) [reply plugins/techdocs/src/home/components/Tables/columns.tsx:35]: Mostly. It's meant to be a duplicate of `createNameColumn` but will generally be hidden

**Comment by squid-ney** [comment]
This looks good, thank so much for contributing a fix! 
One small thing, I don't think it is necessary to add another example component to `techdocs-backend/examples`. From what I see, the only main difference this new example and the existing example component is that the title is different from the component name in the `catalog-info.yaml`. Let me know if I missed something, but based on this, I think you could remove `documented-component-with-title` from the PR.

**Comment by Wilson-StephenA** [comment]
reverted the docs

---

Note: the husky pre-commit failed to work in my environment. It was giving the error:
```txt
".age Error: Couldn't find a script named "lint-staged

$ yarn run [--inspect] [--inspect-brk] [-T,--top-level] [-B,--binaries-only] [--require #0] <scriptName> ...
husky - pre-commit script failed (code 1)
```

checking the script with the following, I was able to confirm that husky was running the right version of yarn in the right working directory, but it was still throwing the error:
```sh
echo $(which yarn)
echo $(pwd)
echo $(yarn --version)
echo $(node --version)
yarn lint-staged
```

**Review by squid-ney — APPROVED** [review]
Thanks for your contribution!

**Comment by squid-ney** [comment]
Sorry for the delay I was on vacation last week. I approved and am good to merge your PR. Wanted to check first though since I see you requested some other folks for review, would you like to wait for their feedback or can I proceed with merging?

**Comment by Zeragamba** [comment]
those reviews were auto requested, so it's up to you/them

**Comment by Wilson-StephenA** [comment]
those reviewers were auto-requested, so it's up to you/them

**Comment by Wilson-StephenA** [comment]
fixed the merge conflict, and changed `createTitleColumn` to just be a wrapper around `createNameColumn`. That way any changes (such as custom sorting) are also applied to the title field

**Comment by CptnFizzbin** [comment]
checking if there's anything more needed on my end for this.

**Pull request merged**